### PR TITLE
Update Filter Implementation in DG

### LINKED ISSUE #193: DG Update Filter Implementation
Describe the implementation of the filter transaction enhancement

### PR DESCRIPTION
**Pull request by jicsontoh**

### CODE REVIEW

**Review by Jo3LW — APPROVED** [review]
LGTM!

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -238,7 +238,56 @@ _{more aspects and alternatives to be added}_
...
+
+`FilterTransCommand` implements the `FilterTransCommand#execute()` operation which executes the command and returns the result message in a `CommandResult` object.
+
+The operation is exposed in the `Logic` interface as `Logic#execute()`.
+
+Given below is an example usage scenario and how the filter mechanism behaves at each step.
+
+Step 1. The user launches the application. The `UiManager` will call on the `MainWindow` to invoke the UI which displays the clients.
+
+![FilterTransState0](images/FilterTransState0png.png)
```
Comment (Jo3LW) [inline docs/DeveloperGuide.md]: Everything looks great! Perhaps renaming the png image file from FilterTransState0png.png to just FilterTransState0.png would be cleaner :)

Comment (jicsontoh) [reply docs/DeveloperGuide.md]: sure, thank you

**Review by ElginL — APPROVED** [review]
LGTM!

**Pull request merged**